zypp/solver/detail/Resolver.cc

```diff
@@ -100,15 +112,42 @@ Resolver::~Resolver()
 sat::detail::CSolver * Resolver::get() const
 { return _satResolver->get(); }
 
+
+void Resolver::setDefaultSolverFlags( bool all_r )
+{
+  MIL << "setDefaultSolverFlags all=" << all_r << endl;
+
+  if ( all_r || _applyDefault_focus ) setFocus( ResolverFocus::Default );
+
+#define ZOLV_FLAG_DEFAULT( ISDEFAULT, ZSETTER )        \
+  if ( all_r || ISDEFAULT ) ZSETTER( indeterminate )
```

Comment:
Using template functions are cleaner than defines. We should consider using those if possible

Reply:
This way It's in sync with the setter/getter defines. It strengthens the desire that all these bit-Flags should follow the same schema.  I would agree if such a macro was used at various locations throughout the code, but it's a continuous block and IMO leaves litte room for misuse.
```
#define ZOLV_FLAG_DEFAULT( ISDEFAULT, ZSETTER )        \
  if ( all_r || ISDEFAULT ) ZSETTER( indeterminate )

  ZOLV_FLAG_DEFAULT( _applyDefault_forceResolve,         setForceResolve         );
  ...
  ZOLV_FLAG_DEFAULT( _applyDefault_dupAllowVendorChange, dupSetAllowVendorChange );
#undef ZOLV_FLAG_TRIBOOL
```